### ISSUE TITLE
Scantailor fails to build

### ISSUE THREAD

**Issue by danstender** [scantailor/scantailor#165]
Hi Nate,

there's a build problem with Scantailor in current Debian Testing:

<cut>
[ 93%] Building CXX object math/spfit/tests/CMakeFiles/spfit_tests.dir/TestSqDistApproximant.cpp.o
Linking CXX executable spfit_tests
CMakeFiles/spfit_tests.dir/TestSqDistApproximant.cpp.o: In function `boost::unit_test::make_test_case(boost::unit_test::callback0<boost::unit_test::ut_detail::unused> const&, boost::unit_test::basic_cstring<char const>)':
TestSqDistApproximant.cpp:(.text._ZN5boost9unit_test14make_test_caseERKNS0_9callback0INS0_9ut_detail6unusedEEENS0_13basic_cstringIKcEE[_ZN5boost9unit_test14make_test_caseERKNS0_9callback0INS0_9ut_detail6unusedEEENS0_13basic_cstringIKcEE]+0x11): undefined reference to `boost::unit_test::ut_detail::normalize_test_case_name[abi:cxx11](boost::unit_test::basic_cstring<char const>)'
collect2: error: ld returned 1 exit status
math/spfit/tests/CMakeFiles/spfit_tests.dir/build.make:118: recipe for target 'math/spfit/tests/spfit_tests' failed
make[2]: *** [math/spfit/tests/spfit_tests] Error 1
CMakeFiles/Makefile2:486: recipe for target 'math/spfit/tests/CMakeFiles/spfit_tests.dir/all' failed
make[1]: *** [math/spfit/tests/CMakeFiles/spfit_tests.dir/all] Error 2
Makefile:146: recipe for target 'all' failed
make: *** [all] Error 2
</cut>

That might be related to this build break:
https://reproducible.debian.net/rb-pkg/unstable/amd64/scantailor.html

Greetings,
Daniel

-- System Information:
Debian Release: stretch/sid
  APT prefers testing
  APT policy: (500, 'testing')
Architecture: amd64 (x86_64)

Kernel: Linux 4.1.0-2-amd64 (SMP w/4 CPU cores)
Locale: LANG=de_DE.utf8, LC_CTYPE=de_DE.utf8 (charmap=UTF-8)
Shell: /bin/sh linked to /bin/dash
Init: systemd (via /run/systemd/system)

Versions of packages scantailor depends on:
ii  dpkg             1.18.2
ii  libc6            2.19-19
ii  libgcc1          1:5.2.1-16
ii  libjpeg62-turbo  1:1.4.1-2
ii  libpng12-0       1.2.50-2+b2
ii  libqt4-xml       4:4.8.7+dfsg-3
ii  libqtcore4       4:4.8.7+dfsg-3
ii  libqtgui4        4:4.8.7+dfsg-3
ii  libstdc++6       5.2.1-16
ii  libtiff5         4.0.5-1
ii  libxrender1      1:0.9.8-1+b1

**Comment by danstender** [comment]
Some progress, the build break referred to in the build log cut above comes from when the Boost unittest framework isn't rebuild against latest gcc5/libstdc++ (dual ABI), this could be fixed by putting -D_GLIBCXX_USE_CXX11_ABI=0 to the build flags. The build break in current Debian appears to be something different.

**Comment by danstender** [comment]
No, there's a problem with Qt4 (4.8.7) in connection with Libboost (1.58), please cf. https://bugs.debian.org/798049 for details

**Comment by danstender** [comment]
We've solved this with scantailor/0.9.11.1-4 (fix-qt4-parsing-error-on-libboost.patch) [1].

Best,
Daniel Stender

[1] http://anonscm.debian.org/cgit/collab-maint/scantailor.git/commit/?id=7a78261

**Comment by ncraun** [comment]
Hi Daniel, thanks for taking charge on this! If you have time, I was wondering if you could make a pull request with this patch, and I will merge it into the repo.

**Comment by danstender** [comment]
Yes, sure, I could prepare this. It'll come up ...

Cheers,
Daniel
